pkg/controller/replication/replication_controller.go

```diff
@@ -654,7 +658,7 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	// modify them, you need to copy it first.
 	// TODO: Do the List and Filter in a single pass, or use an index.
 	var filteredPods []*api.Pod
-	if rm.garbageCollectorEnabled {
+	if rm.garbageCollectorEnabled && !rm.isZeroReplicas(&rc) {
```

Comment:
Since I am not so familiar with the GC. In addition, the ``NewPodControllerRefManager`` does not handle failed pods.
At the same time zero replicas means I want to delete all pods no matter what their state is. Excluding completed pods which are out of any RC.